src/common/webdav.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) The Arvados Authors. All rights reserved.
+//
+// SPDX-License-Identifier: AGPL-3.0
+
+export class WebDAV {
+    static create(config?: Partial<WebDAVDefaults>, createRequest?: () => XMLHttpRequest) {
+        return new WebDAV(config, createRequest);
+    }
+
```

Comment:
Let's replace this static create with normal constructor. This static thing gives us nothing.. Maybe if we had some kind of factory then it would make sense but now it's unnecessary complication.

Reply:
I did it to make interface similar to axios. That's the one we're used to.